addons/purchase_mrp/models/purchase.py

```diff
@@ -18,12 +18,18 @@ class PurchaseOrder(models.Model):
     @api.depends('order_line.move_dest_ids.group_id.mrp_production_ids')
     def _compute_mrp_production_count(self):
         for purchase in self:
-            purchase.mrp_production_count = len(purchase.order_line.move_dest_ids.group_id.mrp_production_ids |
-                                                purchase.order_line.move_ids.move_dest_ids.group_id.mrp_production_ids)
+            purchase.mrp_production_count = len(purchase._get_mrp_productions())
+
+    def _get_mrp_productions(self, allow_archived_picking_types=False):
+        productions = self.order_line.move_dest_ids.group_id.mrp_production_ids
+        productions |= self.order_line.move_ids.move_dest_ids.group_id.mrp_production_ids
+        if not allow_archived_picking_types:
+            productions = productions.filtered(lambda p: p.picking_type_id.active)
```

Comment:
It seems all this code in purchase should be moved in the dropship commit

Reply:
commits adapted